dbcon/joblist/tupleaggregatestep.cpp

```diff
@@ -1782,12 +1782,18 @@ void TupleAggregateStep::prep1PhaseDistinctAggregate(
         }
 
         // vectors for aggregate functions
+        RowAggFunctionType aggOp = ROWAGG_FUNCT_UNDEFINE;
+        RowAggFunctionType prevAggOp = ROWAGG_FUNCT_UNDEFINE;
         for (uint64_t i = 0; i < aggColVec.size(); i++)
         {
             pUDAFFunc = NULL;
             uint32_t aggKey = aggColVec[i].first;
-            RowAggFunctionType aggOp = functionIdMap(aggColVec[i].second);
-            RowAggFunctionType stats = statsFuncIdMap(aggColVec[i].second);
+            aggOp = functionIdMap(aggColVec[i].second);
+            RowAggFunctionType stats = statsFuncIdMap(aggOp);
```

Comment:
Shouldn't we use `aggColVec[i].second` instead of `aggOp`? Looks like that `execplan::AggregateColumn::AggOp` and `rowgroup::RowAggFunctionType` have some differences.

Reply:
Stupid attempt at optimization. Thanks for catching it.